doc/openapi/v0/cyphernode-api.yaml

```diff
@@ -1607,6 +1736,141 @@ components:
           type: "string"
         pub32_index:
           type: "string"
+    UnusedWatchXPubAddress:
+      type: "object"
+      required:
+        - "pub32_watch_id"
+        - "pub32_label"
+        - "pub32"
+        - "address_pub32_index"
+        - "address"
+      properties:
+        pub32_watch_id:
+          type: "string"
+        address:
+          $ref: '#/components/schemas/TypeAddressString'
+        pub32:
+          $ref: '#/components/schemas/TypeXpubString'
+        pub32_label:
+          type: "string"
+        address_pub32_index:
+          type: "string"
+    TransactionsSpending:
+      type: "object"
+      required:
+        - "address"
+        - "category"
+        - "amount"
+        - "label"
+        - "vout"
+        - "fee"
+        - "confirmations"
+        - "blockhash"
+        - "blockindex"
+        - "blocktime"
+        - "txid"
+        - "time"
+        - "timereceived"
+        - "comment"
+        - "bip125-replaceable"
+        - "abandoned"
+      properties:
+        address:
+          $ref: '#/components/schemas/TypeAddressString'
+        category:
+          $ref: 'string'
+        amount:
+          type: "number"
+        label:
+          type: "string"
+        vout:
+          type: "integer"
+        fee:
+          type: "fee"
+        confirmations:
+          type: "integer"
+        blockhash:
+          $ref: '#/components/schemas/TypeHashString'
+        blockindex:
+          $ref: 'number'
+        blocktime:
+          type: "number"
+        txid:
+          $ref: '#/components/schemas/TypeHashString'
+        time:
+          type: "number"
+        timereceived:
+          type: "number"
+        comment:
+          type: "string"
+        bip125-replaceable:
+          type: "string"
+        abandoned:
+          type: "boolean"
+    WatchXPubTxn:
+      type: "object"
+      required:
+        - "label"
+        - "address"
+        - "txid"
+        - "confirmations"
+        - "blockheight"
+        - "v_out"
+        - "amount"
+        - "blockhash"
+        - "blocktime"
+        - "timereceived"
+      properties:
+        label:
+          type: "string"
+        address:
+          $ref: '#/components/schemas/TypeAddressString'
+        txid:
+          $ref: '#/components/schemas/TypeHashString'
+        confirmations:
+          type: "integer"
+        v_out:
+          type: "integer"
+        amount:
+          type: "number"
+        blockhash:
+          $ref: '#/components/schemas/TypeHashString'
+        blocktime:
+          type: "number"
+        timereceived:
+          type: "number"
+    WatchXPubTxn:
```

Comment:
Why is this object defined two times? 
Isn't it just a regular TX definition?
Could you use the Transaction as a base class to WatchXPubTxn and TransactionsSpending, normalising the properties?
Feel free to extend Transaction so inheriting objects have no property duplication
https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/
:-)

Reply:
hmm.. this shouldn't be defined two times. Probably a  coding error on my behalf, will check.
WatchXPubTxn come from sqllite and not from bitcoin core. I believe not all fields of the transaction are saved but will double check and if possible extend what exists. Thanks for the pointer !:)